pkg/provisioningv2/rke2/configserver/server.go

```diff
@@ -116,11 +117,16 @@ func (r *RKE2ConfigServer) connectAgent(planSecret string, secret *v1.Secret, rw
 	}
 
 	if url == "" {
-		_, pem = settings.InternalServerURL.Get(), settings.InternalCACerts.Get()
+		pem = settings.InternalCACerts.Get()
 		url = fmt.Sprintf("https://%s", req.Host)
 		if strings.TrimSpace(pem) != "" {
 			ca = []byte(pem)
 		}
+	} else if v, ok := req.Context().Value(tls.InternalAPI).(bool); ok && v {
```

Comment:
For reviewers: this change is copied from here: https://github.com/rancher/rancher/blob/60275b1f66d2e2c0ae40a9edaa7ca874ac2c23b6/pkg/cacerts/handler.go#L36